src/panels/lovelace/cards/hui-media-control-card.ts

```diff
@@ -310,50 +445,125 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
     });
   }
 
+  private _setColors(): void {
+    if (!this._image) {
+      return;
+    }
+
+    const image: string | HTMLImageElement = this._image;
+
+    Vibrant.from(image)
+      .quality(1)
+      .getPalette()
+      .then((palette: Palette) => {
+        if (!palette.DarkMuted) {
+          this._foregroundColor = undefined;
+          this._backgroundColor = undefined;
+          return;
+        }
+
+        this._backgroundColor = palette.DarkMuted.getHex();
+
+        if (
+          !palette.Vibrant ||
+          getContrastRatio(
+            palette.Vibrant.getRgb(),
+            palette.DarkMuted.getRgb()
+          ) < CONTRAST_RATIO
+        ) {
+          this._foregroundColor = palette.DarkMuted.getBodyTextColor();
+          return;
+        }
+
+        this._foregroundColor = palette.Vibrant.getHex();
+      })
+      .catch((err: any) => {
+        // tslint:disable-next-line:no-console
+        console.error("Error getting Image Colors", err);
+        this._foregroundColor = undefined;
+        this._backgroundColor = undefined;
+      });
+  }
+
+  private _getProgressValue(): void {
+    if (!this._showProgressBar) {
+      this._progressInterval = undefined;
+      this._progressValue = undefined;
+      return;
+    }
+
+    this._progressValue = getCurrentProgress(
```

Comment:
Instead of calculating it here and storing it as an instance variable, just calculate it only in the `render` method. Instead have your interval callback be `() => this.requestUpdate()`

Reply:
hmm without an instance variable how do I only calculate it when its playing. If I do it inline then it calculates everytime it renders. If I calculate in a var before the return template then I need to know the previous value because if I pause the player then I don't want the value to be undefined

Reply:
Actually I think limiting this in the updated is best

Reply:
You just calculate it inside the render to show how far the bar has progressed. You don't base that on an instance variable.

The update interval should only be enabled if the media player is playing.

The update interval should call `this.requestUpdate()`.

Reply:
Yep. Okay. Thats how I ended up doing it :)